content/blog/2023-12-11-bloke/index.md

```diff
@@ -0,0 +1,372 @@
++++
+title = "BLOKE: Optimizing Bril with STOKE"
+[extra]
+bio = """
+    Kei Imada is a first-year Ph.D. student. He is interested in applying
+    mathematical structures to type systems and formal verification to
+    enable efficient development of parallel and distributed systems, and software
+    defined networks.
+"""
+[[extra.authors]]
+name = "Kei Imada"
+link = "https://keikun555.github.io/"
++++
+
+# BLOKE: Optimizing BRIL with STOKE 
+[Presentation](assets/presentation.pdf)
+[Report](assets/report.pdf)
+
+## Abstract
+
+We present bloke, a scalable implementation of STOKE on an educational intermediate representation language. In contrast to STOKE's two-stage process, BLOKE supports a user-defined number of phases that smoothly transitions from synthesis to optimization. We found that BLOKE was able to find superoptimizations of loop-free code written in Bril. Furthermore, we found BLOKE to be scalable with respect to the number of cores a machine has.
+
+## Introduction
+
+Compiler optimizations usually seek to improve a metric on a program. Dead code elimination decreases the number of instructions of programs and loop invariant code motion aims to decrease the number of instructions executed within a loop.
+
+Contrary to the name "optimizations," these algorithms don't seek to optimize, or find the best possible program for the metric. Given this misnomer, Alexia Massalin coined the term superoptimizers.
+
+Given an instruction set, the superoptimizer finds the shortest program to compute a function. [3]
```

Comment:
For a blog post, instead of using footnote citations like `[3]`, you can just link directly to the paper.